src/screens/Profiles/UserLibrary/components/UserLibraryEditScreen/component.js

```diff
@@ -145,169 +154,203 @@ export class UserLibraryEditScreenComponent extends React.Component {
+                ratingTwenty={this.state.ratingTwenty}
+                ratingSystem={ratingSystem}
+              />
+            </View>
+            {/* Rewatched */}
+            <View style={styles.editRow}>
+              <Text style={styles.editRowLabel}>
+                Times {libraryType === 'anime' ? 'Rewatched' : 'Read'}:
               </Text>
-              <TextInput
-                style={[styles.editRowValue, { height: this.state.notesHeight }]}
-                value={this.state.notes}
-                placeholder={canEdit ? "Type some notes..." : "User has no notes"}
-                onChangeText={this.onNotesChanged}
-                onContentSizeChange={this.onNotesContentSizeChanged}
-                editable={canEdit}
-                multiline
+              <Counter
+                inputRef={(r) => { this.rewatchInput = r; }}
+                disabled={!canEdit}
+                initialValue={libraryEntry.reconsumeCount}
+                onValueChanged={this.onTimesRewatchedChanged}
               />
             </View>
-          </View>
-          {/* Delete */}
-          {canEdit &&
-            <View style={[styles.deleteEntryRow]}>
-              <SelectMenu options={['Yes, Delete.', 'Nevermind']} onOptionSelected={this.onDeleteEntry}>
-                <Text style={styles.deleteEntryText}>Delete Library Entry</Text>
+            {/* Visibility */}
+            <View style={styles.editRow}>
+              <Text style={styles.editRowLabel}>
+                Library Entry Visibility:
+              </Text>
+              <SelectMenu
+                disabled={!canEdit}
+                options={visibilityOptions}
+                onOptionSelected={this.onVisibilityChanged}
+              >
+                <Text style={styles.editRowValue}>
+                  {this.state.private ? 'Private' : 'Public'}
+                </Text>
               </SelectMenu>
             </View>
-          }
-        </KeyboardAwareScrollView>
+            {/* Dates */}
+            {/* @TODO: Needs to be editable */}
+            <View style={styles.splitRow}>
+              <View style={[styles.editRow, styles.dateStarted]}>
+                <View>
+                  <Text style={[
+                    styles.editRowLabel,
+                    this.state.startedAt && styles.withValueLabel,
+                  ]}
+                  >
+                    Date Started
+                  </Text>
+                  {this.state.startedAt &&
+                    <Text style={styles.editRowValue}>
+                      {moment(this.state.startedAt).format('MMM. DD, YYYY')}
+                    </Text>
+                  }
+                </View>
+              </View>
+              <View style={[styles.editRow, styles.dateFinished]}>
+                <View>
+                  <Text style={[
+                    styles.editRowLabel,
+                    this.state.finishedAt && styles.withValueLabel,
```

Comment:
we can deconstruct `finishedAt` and `startedAt` before rendering to improve readability:

`const { finishedAt } = this.state` like you did above.